### ISSUE TITLE
gffread does not seem to use the .fai files it creates.

### ISSUE THREAD

**Issue by NDHall** [gpertea/gffread#59]
`gffread v0.12.2` fails to use a preexisting `.fai` file it created in a previous run. The first run in which there no `.fai` file `gffread` succeeds in producing the correct output.

* Here is second run in a series of feature extractions, the first run I did was with `-x` flag. Just to be clear, I never indexed these files with `samtools faidx` so all `.fai` were created by `gffread`.
```
$  gffread ../gff/Atriplex_hortensis_56906.gff  -g ../genomes/Atriplex_hortensis_56906.faa  -y Atriplex_hortensis_56906_prots.fna 
Warning: duplicate sequence ID (CAF_434) added to the fasta index! Only last entry data will be kept.
Warning: duplicate sequence ID (CAF_434) added to the fasta index! Only last entry data will be kept.
Warning: duplicate sequence ID (CAF_434) added to the fasta index! Only last entry data will be kept.
Warning: duplicate sequence ID (��) added to the fasta index! Only last entry data will be kept.
Warning: duplicate sequence ID (��) added to the fasta index! Only last entry data will be kept.
Warning: duplicate sequence ID (��) added to the fasta index! Only last entry data will be kept.
Warning: duplicate sequence ID (��) added to the fasta index! Only last entry data will be kept.
Warning: duplicate sequence ID (ȷ;�) added to the fasta index! Only last entry data will be kept.
Warning: duplicate sequence ID (���) added to the fasta index! Only last entry data will be kept.
Warning: duplicate sequence ID (���) added to the fasta index! Only last entry data will be kept.
Warning: duplicate sequence ID (���) added to the fasta index! Only last entry data will be kept.
Warning: duplicate sequence ID (���) added to the fasta index! Only last entry data will be kept.
Warning: duplicate sequence ID (���) added to the fasta index! Only last entry data will be kept.
Warning: duplicate sequence ID (���) added to the fasta index! Only last entry data will be kept.
...
Warning: duplicate sequence ID (ط;�) added to the fasta index! Only last entry data will be kept.
Warning: duplicate sequence ID (ط;�) added to the fasta index! Only last entry data will be kept.
Warning: duplicate sequence ID (ط;�) added to the fasta index! Only last entry data will be kept.
Warning: duplicate sequence ID (ط;�) added to the fasta index! Only last entry data will be kept.
Warning: duplicate sequence ID (ط;�) added to the fasta index! Only last entry data will be kept.
Warning: duplicate sequence ID (ط;�) added to the fasta index! Only last entry data will be kept.
Warning: duplicate sequence ID (ط;�) added to the fasta index! Only last entry data will be kept.
Warning: duplicate sequence ID (ط;�) added to the fasta index! Only last entry data will be kept.
Warning: couldn't find fasta record for 'Scaffold_1014_HRSCAF_1334'!
Error: no genomic sequence available (check -g option!).

$ egrep Scaffold_1014_HRSCAF_1334 ../genomes/Atriplex_hortensis_56906.faa
>Scaffold_1014_HRSCAF_1334
# no white space in header.
```
*  egrep indicates that the header is in fact in the file. It also appears to be free of any white space.
* I remove the `.fai` file and re-run the command, and it produces the desired output. I had to do this with every one of the gff/genome file pairs I was using. 
``` 
$ rm ../genomes/Atriplex_hortensis_56906.faa.fai 
$  gffread Atriplex_hortensis_56906.gff  -g ../genomes/Atriplex_hortensis_56906.faa  -y Atriplex_hortensis_56906_prots.fna 
FASTA index file ../genomes/Atriplex_hortensis_56906.faa.fai created.

$ head Atriplex_hortensis_56906_prots.fna
>Ah039314.mRNA1
MSSGPHMFNGRAAINRVPLWVKFPNLPLNWPGSETLSRIASLLGQLRVSFALVPVEIDVTKPLQHSVWIQ
SPAGEQMELKVQYEWTPPFCAKCQKVGHDCNEKEKTPKEVPKDRKPQKAQKTHNPQKPQKQVWVQKAPEQ
QDSSETINPIMNPPEQWNIVTRKKKSYASVQRNDVHQTNSFDALAEDTEANEIRNNNTVEPSGLQRRIWL
GWRDSRVTVDILASHEQFVHTTISNVDGSHSTSCTFVYGLNTTQDRLPLGRVEKLSSSDPLGPYCLAPRL
CSVVAIYLPGLSDHNPMLLEIVEIILQKGLKALNVKHYSTIDKEIEEAYNNLDQIMLYPRIFQIAELHTQ
EVDAIKNLRFG
>Ah039372.mRNA1
MVAIPNGDLASILDYRHAQKWVRSNVVRYKFKGGVNINKMFEIGGLSGNCRGFNEVEAAVMIEQNSRWVI
VLIVEGGKVSKDRKVGLLLLKLFFWCRKAEKTYDNIALPTAFFFLARLYVAVGNEPFLKAYNGTFEKATL

```

**Comment by gpertea** [comment]
wow, this is a terrible bug -- the problem is that the created fai file is actually broken (the sequence names are corrupted).  That's what I get if I don't have a proper test suite :). Thank you so much for reporting this ugly bug, I'll fix it ASAP.

**Comment by NDHall** [comment]
Thanks for looking into this
-N
________________________________

**Comment by gpertea** [comment]
This should be fixed now in the latest v0.12.3 release.
